presentation-compiler/src/main/scala/org/alephium/ralph/lsp/pc/workspace/WorkspaceSearcher.scala

```diff
@@ -153,19 +153,25 @@ object WorkspaceSearcher {
    * @param workspace The workspace to search for types.
    * @return An iterator containing type identifiers.
    */
-  def collectTypes(workspace: WorkspaceState.IsSourceAware): Iterator[SourceLocation.Node[Ast.TypeId]] = {
-    val trees = collectTrees(workspace)
+  def collectTypes(
+      workspace: WorkspaceState.IsSourceAware,
+      includeNonImportedCode: Boolean): Iterator[SourceLocation.Node[Ast.TypeId]] = {
+    val trees = collectTrees(workspace, includeNonImportedCode)
     SourceCodeSearcher.collectTypes(trees.iterator)
   }
 
   /**
    * Collects all parsed source files, excluding `std` dependency source files
    * that are not imported.
    *
-   * @param workspace The workspace with dependencies.
+   * @param workspace              The workspace with dependencies.
+   * @param includeNonImportedCode If true, includes dependency code that is not imported,
+   *                               otherwise, excludes non-imported dependency code.
```

Comment:
Just to make sure I understand well
>  dependency code that is not imported

it's code coming from `extends` etc?

vs imported code which comes from the `import "std/..` ?

And why do we sometimes excludes them?

Reply:
> it's code coming from extends etc?

All types in the workspace, extended or not, are all suggested, not conditional. Similar behaviour to IntelliJ. But IntelliJ also bring in the import statement automatically. 

> And why do we sometimes excludes them?

They excluded from go-to definition when the import statement is missing. Similar to IntelliJ.